### PR TITLE
Add ($) operator

### PR DESCRIPTION
**Pull request by xsebek**
- add `$` operator to avoid multiple parentheses

### CODE REVIEW

**Review by byorgey — APPROVED** [review]
At first I thought doing it as syntactic sugar wouldn't work---but in fact since we don't support operator sections or `($)` notation at all, it should work fine.

Actually `x,y,z` isn't syntactic sugar, that's just how it parses (though we actually need to fix that, see #225).

I don't have any good ideas re: what could provide this capability.  We might just make it free, since it doesn't seem like it's going to be very worthwhile to harvest stuff just to make a device that lets you avoid writing a few parentheses.  Hmm, but then users will never discover it since it won't be in the description of any entity...

File: src/Swarm/Language/Syntax.hs [hunk 1]
```diff
@@ -424,6 +429,7 @@ constInfo c = case c of
   Gt -> binaryOp ">" 4 N
   Leq -> binaryOp "<=" 4 N
   Geq -> binaryOp ">=" 4 N
+  AppF -> binaryOp "$" 0 R
```
[the remark this answers is not in the document]
Comment (xsebek) [reply src/Swarm/Language/Syntax.hs:432]: I read a bit and it seems quite specific to Haskell in that everyone there can define whatever operator they wish.
We do not have `(.)` so the fact  that `($)` can be used instead sometimes is actually good for us :slightly_smiling_face:

**Comment by xsebek** [comment]
A funny thing is that... ehm... no entity can be required for `$`, as it is gone before we check for it. :sweat_smile: 

How will users discover this entity then? :thinking:

**Comment by xsebek** [comment]
@byorgey actually I might have been wrong - in the documeation it reads:
> Given a 'Text' value representing a Swarm program,
> 1. Parse it (see "Swarm.Language.Parse")
> 2. Typecheck it (see "Swarm.Language.Typecheck")
> 3. Elaborate it (see "Swarm.Language.Elaborate")
> 4. Check what capabilities it requires (see "Swarm.Language.Capability")

But later in the code:
```haskell
processParsedTerm' ctx capCtx t = do
  ty <- inferTop ctx t
  let (caps, capCtx') = requiredCaps capCtx (sTerm t)
  return $ ProcessedTerm (elaborate (sTerm t)) ty caps capCtx'
```
Which seems like we already do 4. before 3. and it may be #231 that is the issue.

**Comment by xsebek** [comment]
Personally I would be fine if `processParsedTerm'` triggered the required check on base - there is no need to install anything on robots for `$`. Unless there is an easy way to fix #231 here, I would go ahead and merge.

**Comment by byorgey** [comment]
Yeah, I am fine if `$` is free.  I don't think #231 will be hard to fix but we probably shouldn't do it here.

**Pull request merged**